### PR TITLE
Fully Recurring Events

### PR DESCRIPTION
**Pull request by smazik02**
Added a full event recurrence with all required REST endpoints, including fetching user's events with all exceptions and ocurring dates between two dates given in a query search.
Also added an URL that 'seeds' the database with all necessary data without needing to apply a migration (might delete it if you want to, this was just to make testing easier and quicker)

### CODE REVIEW

File: src/app.controller.ts [hunk 1]
```diff
@@ -9,4 +9,9 @@ export class AppController {
     getHello(): string {
         return this.appService.getHello();
     }
+
+    @Get('seed')
+    async seedDatabase() {
+        return this.appService.seedDatabase();
+    }
```
Comment (Aeri4a) [inline src/app.controller.ts:16]: To remove.

File: src/app.module.ts [hunk 2]
```diff
@@ -26,14 +30,20 @@ import { SearchModule } from './resource/search/search.module';
             }),
             inject: [ConfigService],
         }),
+        TypeOrmModule.forFeature([
+            UserEntity,
+            EventEntity,
+            EventExceptionEntity,
+            RecurrenceEntity,
+        ]),
```
Comment (Aeri4a) [inline src/app.module.ts:39]: To remove 2.

File: src/app.service.ts
Comment (Aeri4a) [inline src/app.service.ts:1]: To remove 3.

File: src/resource/event/event.controller.ts [hunk 3]
```diff
@@ -8,66 +8,133 @@ import {
...
 
     @Get(':id')
     @GetEventByIdApi()
-    getEventById(@Param('id') eventId: number): Promise<ReturnEventDTO> {
-        return this.eventService.getEventById(eventId);
+    async getEventById(@Param('id') eventId: number): Promise<ReturnEventDTO> {
+        return await this.eventService.fetchById(eventId);
+    }
+
+    @Get('except/:id')
```
Comment (Aeri4a) [inline src/resource/event/event.controller.ts:80]: exception

File: src/resource/event/event.service.ts [hunk 4]
```diff
@@ -1,28 +1,181 @@
...
         if (!event) throw new NotFoundException('Event not found');
         return event;
     }
 
-    async getEventsByUserId(userId: number): Promise<EventEntity[]> {
-        return this.eventRepository.findBy({ createdById: userId });
+    async fetchUserEvents(userId: number): Promise<ReturnEventDTO[]> {
+        return this.eventRepository.find({
+            where: { createdById: userId },
+            relations: { recurrencePattern: true, eventExceptions: true },
```
Comment (Aeri4a) [inline src/resource/event/event.service.ts:42]: relations: ['recurrencePattern', 'eventExceptions']

File: src/resource/event/event.service.ts [hunk 5]
```diff
@@ -1,28 +1,181 @@
...
+        userId: number,
+        startDate: Date,
+        endDate: Date,
+    ): Promise<ReturnEventWithDatesDTO[]> {
+        const toFilter = await this.eventRepository.find({
+            where: {
+                createdById: userId,
+                startDate: Raw(
+                    (alias) => `${alias} <= DATE ('${endDate.toISOString()})')`,
+                ),
```
Comment (Aeri4a) [inline src/resource/event/event.service.ts:56]: Less/More/Greater sth like this to replace

File: src/resource/event/event.service.ts [hunk 6]
```diff
@@ -33,27 +186,66 @@ export class EventService {
...
+        const exceptionToUpdate = await this.exceptionRepository.findOneBy({
+            id: exceptionId,
+        });
+        if (!exceptionToUpdate)
+            throw new NotFoundException('Exception not found');
+        this.exceptionRepository.merge(exceptionToUpdate, editExceptionDto);
+        await this.eventRepository.update(
+            { id: exceptionToUpdate.id },
+            { ...editExceptionDto },
+        );
```
Comment (Aeri4a) [inline src/resource/event/event.service.ts:240]: check if is not redundant

**Review by Aeri4a — APPROVED** [review]

File: src/app.controller.ts (hunk 1 above)
Comment (smazik02) [reply src/app.controller.ts:16]: done

**Pull request merged**